identity/app/clients/DiscussionClient.scala

```diff
@@ -82,4 +94,29 @@ class DiscussionClient(wsClient: WSClient, config: conf.IdentityConfiguration)(i
       case e => Future.failed(new DiscussionApiServiceException(s"findProfileStats failure to find $userId", e))
     }
   }
+
+  /**
+    * Check if the user has at least one comment. This is a lightweight check which replaces
+    * the findProfileStats call, which has been found to be extremely slow for users with a large
+    * number of comments, as on the Discussion API side it involves a full count of all comments.
+    * This method simply requests a single comment, and checks if the response is non-empty.
+    * If it is, the user has at least one publicly visible comment, and their profile can be shown.
+    */
+  def profileHasAtLeastOneComment(userId: String): Future[Boolean] = {
+
+    val apiPath = s"/profile/$userId/comments?page=1&pageSize=1"
+
+    val discussionCommentsF = for {
+      response <- GET(apiPath)
+      userCommentsResponse <- Future.fromTry(handleApiResponse(response, userId))
+    } yield userCommentsResponse.map(_.as[DiscussionCommentsResponse].comments.nonEmpty)
+
+    discussionCommentsF.recoverWith {
+      case e =>
+        Future.failed(new DiscussionApiServiceException(s"profileHasAtLeastOneComment failure to find $userId", e))
+    }
+
+    discussionCommentsF.map(_.contains(true))
```

Comment:
Maybe this would be a little easier to understand (If it even works, **I HAVE NOT TESTED IT!!**):
```suggestion
    val hasCommentsF = for {
      response <- GET(apiPath)
      userCommentsResponse <- Future.fromTry(handleApiResponse(response, userId))
      parsedComments = userCommentsResponse.map(_.as[DiscussionCommentsResponse])
    } yield parsedComments.exists(_.comments.nonEmpty)

    hasCommentsF.recoverWith {
      case e =>
        Future.failed(new DiscussionApiServiceException(s"profileHasAtLeastOneComment failure to find $userId", e))
    }
```

Reply:
This works great! Yes, I agree it's easier to understand without the `map` at the very end.